packages/story-editor/src/components/panels/design/textAccessibility/textAccessibility.js

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { __, sprintf, TranslateWithMarkup } from '@googleforcreators/i18n';
+import {
+  DropDown,
+  Text,
+  THEME_CONSTANTS,
+  Link,
+} from '@googleforcreators/design-system';
+import { trackClick } from '@googleforcreators/tracking';
+import { useFeature } from 'flagged';
+import { useCallback, useState } from '@googleforcreators/react';
+import { getTextElementTagNames } from '@googleforcreators/output';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../../app/story';
+import { Row } from '../../../form';
+import { getCommonValue } from '../../shared';
+import { SimplePanel } from '../../panel';
+import { MULTIPLE_VALUE, MULTIPLE_DISPLAY_VALUE } from '../../../../constants';
+
+const HEADING_LEVELS = {
+  h1: __('Heading 1', 'web-stories'),
+  h2: __('Heading 2', 'web-stories'),
+  h3: __('Heading 3', 'web-stories'),
+  p: __('Paragraph', 'web-stories'),
+};
+
+function TextAccessibilityPanel({ selectedElements, pushUpdate }) {
+  const showSemanticHeadings = useFeature('showSemanticHeadings');
+  const [tagNamesOverrides, setTagNamesOverrides] = useState();
+
+  const { textElements } = useStory(({ state }) => ({
+    textElements: state.currentPage?.elements
+      ?.filter(({ type }) => 'text' === type)
+      .map(({ id, tagName, ...rest }) => ({
+        id,
+        ...rest,
+        tagName: tagNamesOverrides?.get(id) || tagName,
+      })),
+  }));
+
+  const onLinkClick = useCallback((evt) => {
+    trackClick(evt, 'click_headings_docs');
+  }, []);
+
+  const handleChange = useCallback(
+    (evt, value) => {
+      setTagNamesOverrides(
+        new Map(selectedElements.map(({ id }) => [id, value]))
+      );
+
+      selectedElements.forEach(() => {
+        pushUpdate({ tagName: value });
```

Comment:
This might need `true` flag here as well to actually push the update. You can see it by having a clean history state and changing the tag -- there's no history entry created. The `pushUpdate` system needs updating (or removing), it's a bit out of date, there's an issue for it, too: #6690